vscode/src/debugger/session.ts

```diff
@@ -874,4 +813,57 @@ export class QscDebugSession extends LoggingDebugSession {
     );
     this.sendEvent(evt);
   }
+
+  /**
+   * Attempts to find the Source in the current session and returns the
+   * TextDocument if it exists.
+   *
+   * This method *may* return a valid result even when the requested
+   * path does not belong in the current program (e.g. another Q# file
+   * in the workspace).
+   */
+  async tryLoadSource(source: DebugProtocol.Source) {
+    if (!source.path) {
+      return;
+    }
+
+    const uri = this.asUri(source.path);
+    if (!uri) {
+      return;
+    }
+
+    try {
+      const doc = await vscode.workspace.openTextDocument(uri);
```

Comment:
If I'm reading the doc right, this actually opens the document in the editor, correct? Is this the desired behavior for 'tryLoadSource'? It looks like this is called when resolving breakpoints, which happens at the start of the session I think, so does this mean when I start debugging, every doc that has a breakpoint in it will be opened in the editor?

Reply:
No, `openTextDocument` doesn't open a visible text editor for the document (that's `showTextDocument`). Think of it like VS Code opening the document in the background - it's handy when you need information from the document model but don't want to show UI. We used to use `openTextDocument` all over the place to map offsets to line+columns, without opening the doc (but we don't anymore since lines/columns are mapped in Rust now).